File: SophomoreYear/CMPS101/pa1/Lex.java
//-------------------------------------------------------------------
// Created by:	Bryan Lopez
// 		blopez24
//-------------------------------------------------------------------
// File:	Lex.java
//-------------------------------------------------------------------
import java.io.*;
import java.util.Scanner;


class Lex
{
	public static void main(String[] args) throws IOException
	{
		Scanner in = null;
                PrintWriter out = null;
                int n = 0, lines = 0;

                if(args.length < 2)
                {
                        System.err.println("Usage: Lex <input file> <output file>");
                        System.exit(1);
                }

                in = new Scanner(new File(args[0]));
                out = new PrintWriter(new FileWriter(args[1]));

                while( in.hasNextLine() )
                {
                        lines = lines + 1;
                        in.nextLine();
                }
                in.close();

                String[]  array = new String[lines];
                in = new Scanner(new File(args[0]));

                while( in.hasNextLine() )
                {
                        array[n] = in.nextLine();
                        n++;
                }	

		List A = new List();
		
		for(int i = 0; i < array.length; i++)
		{
			if(i == 0)
				A.prepend(i);

			else
			{
				String str1 = array[i];
				
				A.moveBack();
				while( A.index() >= 0 )
				{
					String str2 = array[A.get()];
					int compareValue = str1.compareTo(str2);	

					if(A.index() == 0 && compareValue < 0)
					{
						A.insertBefore(i);
						break;
					}
					else if(compareValue < 0)
					{
						A.movePrev();
					}
					else
					{
						A.insertAfter(i);
						break;
					}
				}
			} 
		}
		System.out.println(A);
		for(A.moveFront(); A.index() >= 0; A.moveNext())
		{
			out.println(array[A.get()]);
		}

		in.close();
		out.close();
	}
}

File: SophomoreYear/CMPS101/pa2/Lex.c
//-----------------------------------------------------------------------------
// Created by:	Bryan Lopez
//				blopez24
//-----------------------------------------------------------------------------
// File:		Lex.c
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include "List.h"

#define MAX_LEN 1024

int main(int argc, char * argv[])
{
	int n = 0, count = 0;
	FILE *in, *out;
	char line[MAX_LEN];

	// check command line for correct number of arguments
	if( argc != 3 )
	{
		printf("Usage: %s <input file> <output file>\n", argv[0]);
		exit(1);
	}

	// opens files for reading and writing
	in = fopen(argv[1], "r");
	out = fopen(argv[2], "w");
	
	if( in == NULL )
	{
		printf("Unable to open file %s for reading\n", argv[1]);
		exit(1);
	}
	if( out == NULL )
	{
		printf("Unable to open file %s for writing\n", argv[2]);
		exit(1);
	}

	/* count each line in input file */
	while( fgets(line, MAX_LEN, in) != NULL )
	{
		count++;
	}
	fclose(in);
	
	in = fopen(argv[1], "r");
	
	char words[count][MAX_LEN];
	
	while( fgets(line, MAX_LEN, in) != NULL )
	{
		strcpy(words[n], line);
		n++;
	}

	List A = newList();
	prepend(A, 0);
	
	for(int i = 1; i < n; i++)
	{
		char* str1 = words[i];
		
		moveBack(A);
		while( index(A) >= 0 )
		{
			char* str2 = words[get(A)];
			int compareValue = strcmp(str1, str2);
			
			if( index(A) == 0 && compareValue < 0 )
			{
				insertBefore(A, i);
				break;
			}
			else if( compareValue < 0 )
			{	
				movePrev(A);
			}
			else
			{	
				insertAfter(A, i);
				break;
			}
		}
	}
	
	for(moveFront(A); index(A) >= 0; moveNext(A))
	{
		fprintf(out, "%s", words[get(A)]);
	}
	
	freeList(&A);
	
	/* close files */
	fclose(in);
	fclose(out);

	return(0);
}
